util/parsec_project.py

```
@@ -181,38 +201,51 @@ def get_functions_in_topological_order(self, reverse_order: bool = False) -> lis
         """
-        parsec_executable = os.environ.get("PARSEC_EXECUTABLE")
-        if not parsec_executable:
-            raise Exception("$PARSEC_EXECUTABLE not set.")
-        try:
-            cmd = f"{parsec_executable} --rename-main=false --add-instr=false {file_path}"
+        if run_mode == ParsecRunMode.FILE:
+            if not path.exists():
+                msg = f"File {path} does not exist"
+                raise FileNotFoundError(msg)
+            cmd = f"parsec --rename-main=false --add-instr=false {path}"
             result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
```

Ruff (0.15.0)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)